### ISSUE TITLE
Doesn't pickup files from the tests folder

### ISSUE THREAD

**Issue by Razzeee** [stil4m/elm-analyse#207]
Probably due to them missing here?
https://github.com/stil4m/elm-analyse/blob/master/ts/util/file-gatherer.ts#L27

**Comment by stil4m** [comment]
Yes. This was originally designed how elm test functioned in `0.18`. It would make more sense to include test files as well at this moment

**Comment by Razzeee** [comment]
so how should we proceed with this?

**Comment by stil4m** [comment]
A change has to be made to make this work.
I propose to extend the mechanism of collecting source files to also collect the test files.
This would only apply if there are dependencies in the `test-dependencies` block (I think that is a sufficient predicate).

I'm not sure how including test files will affect the program. 
For example, a dependency is actually unused if it is only used in test code and not source code. Thus to actually pertain the same properties as the system does now, files should be marked as test files.

If I got time I can do a POC, or feel free to prepare a POC yourself.

**Comment by Razzeee** [comment]
I probably won't have time, have some things I need to do for the elm language server. Which includes upstreaming our elm-analyse fork :)
